Advise to remove files & functions without comments from commit message

annevk opened the pull request:
#### 7a2104ff37a417bae31bf64e479b42e332fab560
<pre>
Advise to remove files &amp; functions without comments from commit message
<a href="https://bugs.webkit.org/show_bug.cgi?id=292534">https://bugs.webkit.org/show_bug.cgi?id=292534</a>

Reviewed by Keith Miller.

In addition, slightly adjust the line length as it was quite narrow.

Canonical link: <a href="https://commits.webkit.org/294775@main">https://commits.webkit.org/294775@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/62111b12a6cf72400db57982af99693cd862bee1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102319 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21988 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12302 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107480 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52955 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104358 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22296 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30494 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77856 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34845 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105325 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17261 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92360 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58194 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/101792 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17095 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10388 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52313 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86923 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10458 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109855 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29451 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21713 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86836 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/101874 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29815 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88556 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86425 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31238 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8953 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23693 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16713 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29379 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29190 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32513 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30749 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Tools/Scripts/hooks/prepare-commit-msg:
```diff
@@ -342,8 +342,9 @@ def main(file_name=None, source=None, sha=None):
-# with '#' will be ignored
+# Please populate the above commit message. Lines starting with '#'
+# will be ignored. If you don't provide per-file-and-function comments,
+# then please remove the list of files and functions.
```
I would say "For any files or functions that don't have an associated comment please remove from the commit message." The current wording implies to me you should only remove the list if you don't have any comments on any file/function.

kmiller68 commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/hooks/prepare-commit-msg:
```diff
@@ -23,8 +23,9 @@ def main(file_name=None, source=None, sha=None):
-# with '#' will be ignored
+# Please populate the above commit message. Lines starting with '#'
+# will be ignored. If you don't provide per-file-and-function comments,
+# then please remove the list of files and functions.
```
Ditto.

kmiller68 commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py:
```diff
@@ -228,8 +228,9 @@ def test_commit_message(self):
-# with '#' will be ignored
+# Please populate the above commit message. Lines starting with '#'
+# will be ignored. If you don't provide per-file-and-function comments,
+# then please remove the list of files and functions.
```
Ditto.

kmiller68 reviewed (approved):
r=me, nice!

annevk committed:
Advise to remove files & functions without comments from commit message
https://bugs.webkit.org/show_bug.cgi?id=292534

Reviewed by Keith Miller.

In addition, slightly adjust the line length as it was quite narrow.

Canonical link: https://commits.webkit.org/294775@main
